packages/graph/hash_graph/lib/graph/src/store/pool.rs

```diff
@@ -6,7 +6,11 @@ use crate::store::Store;
 #[async_trait]
 pub trait StorePool: Send + Sync {
     type Error;
-    type Store: Store + Send;
+    type Store<'pool>: Store + Send + Sync
+    where
+        Self: 'pool;
 
-    async fn acquire(&self) -> Result<Self::Store, Self::Error>;
+    async fn acquire(&self) -> Result<Self::Store<'_>, Self::Error>;
```

Comment:
Could we add some docstrings in this trait? Both for the trait and the methods i presume :)

Reply:
Added a documentation in 7df772798b29cc035ccdb7bb638cbfeb4b4eda83

I created a task to track the documentation: [Asana task](https://app.asana.com/0/1201095311341924/1202668019871915/f) (_internal_)